src/Libraries/Nop.Data/Migrations/UpgradeTo460/DataMigration.cs

```diff
@@ -0,0 +1,104 @@
+                    customer.LastName = customerAttributes.FirstOrDefault(ga => ga.Key == "LastName")?.Value;
+                    customer.Gender = customerAttributes.FirstOrDefault(ga => ga.Key == "Gender")?.Value;
+                    customer.Company = customerAttributes.FirstOrDefault(ga => ga.Key == "Company")?.Value;
+                    customer.StreetAddress = customerAttributes.FirstOrDefault(ga => ga.Key == "StreetAddress")?.Value;
+                    customer.StreetAddress2 = customerAttributes.FirstOrDefault(ga => ga.Key == "StreetAddress2")?.Value;
+                    customer.ZipPostalCode = customerAttributes.FirstOrDefault(ga => ga.Key == "ZipPostalCode")?.Value;
+                    customer.City = customerAttributes.FirstOrDefault(ga => ga.Key == "City")?.Value;
+                    customer.County = customerAttributes.FirstOrDefault(ga => ga.Key == "County")?.Value;
+                    customer.Phone = customerAttributes.FirstOrDefault(ga => ga.Key == "Phone")?.Value;
+                    customer.Fax = customerAttributes.FirstOrDefault(ga => ga.Key == "Fax")?.Value;
+                    customer.VatNumber = customerAttributes.FirstOrDefault(ga => ga.Key == "VatNumber")?.Value;
+                    customer.TimeZoneId = customerAttributes.FirstOrDefault(ga => ga.Key == "TimeZoneId")?.Value;
+                    customer.CustomCustomerAttributesXML = customerAttributes.FirstOrDefault(ga => ga.Key == "CustomCustomerAttributesXML")?.Value;
+
+                    if (int.TryParse(customerAttributes.FirstOrDefault(ga => ga.Key == "CountryId")?.Value, out var countryId))
+                        customer.CountryId = countryId;
+                    if (int.TryParse(customerAttributes.FirstOrDefault(ga => ga.Key == "StateProvinceId")?.Value, out var stateProvinceId))
+                        customer.StateProvinceId = stateProvinceId;
+                    if (int.TryParse(customerAttributes.FirstOrDefault(ga => ga.Key == "VatNumberStatusId")?.Value, out var vatNumberStatusId))
+                        customer.VatNumberStatusId = vatNumberStatusId;
+                    if (int.TryParse(customerAttributes.FirstOrDefault(ga => ga.Key == "CurrencyId")?.Value, out var currencyId))
+                        customer.CurrencyId = currencyId;
+                    if (int.TryParse(customerAttributes.FirstOrDefault(ga => ga.Key == "LanguageId")?.Value, out var languageId))
+                        customer.LanguageId = languageId;
+                    if (int.TryParse(customerAttributes.FirstOrDefault(ga => ga.Key == "TaxDisplayTypeId")?.Value, out var taxDisplayTypeId))
+                        customer.TaxDisplayTypeId = taxDisplayTypeId;
+                    if (DateTime.TryParse(customerAttributes.FirstOrDefault(ga => ga.Key == "DateOfBirth")?.Value, out var dateOfBirth))
+                        customer.DateOfBirth = dateOfBirth;
+
```

Comment:
We still need to delete attributes that have been moved to the table and leave all the others.